File: Data Structures & Algorithms/merge-intervals/submission-11.js
class Solution {
    /**
     * @param {number[][]} intervals
     * @return {number[][]}
     */
    merge(intervals) {
        const mp = new Map();
        for (const [start, end] of intervals) {
            mp.set(start, (mp.get(start) || 0) + 1);
            mp.set(end, (mp.get(end) || 0) - 1);
        }

        const sortedKeys = Array.from(mp.keys()).sort((a, b) => a - b);
        const res = [];
        let interval = [];
        let have = 0;

        for (const i of sortedKeys) {
            if (interval.length === 0) {
                interval.push(i);
            }
            have += mp.get(i);
            if (have === 0) {
                interval.push(i);
                res.push(interval);
                interval = [];
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/minimum-interval-including-query/submission-11.cs
public class Solution {
    public int[] MinInterval(int[][] intervals, int[] queries) {
        int[] res = new int[queries.Length];
        int idx = 0;
        foreach (int q in queries) {
            int cur = -1;
            foreach (var interval in intervals) {
                int l = interval[0], r = interval[1];
                if (l <= q && q <= r) {
                    if (cur == -1 || (r - l + 1) < cur) {
                        cur = r - l + 1;
                    }
                }
            }
            res[idx++] = cur;
        }
        return res;
    }
}